### ISSUE TITLE
Add action for after remove_order_item

### ISSUE THREAD

**Issue by pinksharpii** [woocommerce/woocommerce#26857]
I'm writing some custom functionality for a client that involves some hooks when a product is added or removed from an order within the admin. I was easily able to hook into the AJAX callback for add_order_item [with this hook](https://docs.woocommerce.com/wc-apidocs/source-class-WC_AJAX.html#920).

However, looking at the AJAX callback for [remove_order_item](https://docs.woocommerce.com/wc-apidocs/source-class-WC_AJAX.html#1210) there is not a do_action here that I can hook into and have access to the $order like we can in the AJAX callback for add_order_item above.
This function does call wc_delete_order_item() which has a hook `do_action( 'woocommerce_delete_order_item', $item_id );` but there's no $order/$order_id available there, just the removed item id.

**Comment by ObliviousHarmony** [comment]
Hi @pinksharpii,

Thank you for taking the time to share this idea, we really appreciate your help. 

While this is clearly a great improvement/idea we won’t be able to tackle it in the upcoming weeks. We’re going to add it to our backlog where it will be considered for future releases.

As a way for you to work around the limitations of the current actions, I would suggest taking a look at the `woocommerce_before_delete_order_item` action. Since this takes place before the item is deleted from the database you can still get whatever data you are looking for from it. For instance, if you want to load the object, you can use `new WC_Order_item_Product( $item_id )` (you'll want to check if its a product item first!). If all you care about is finding the order ID, you can do that with `wc_get_order_id_by_order_item_id( $item_id )`.

Stay tuned for updates.

**Comment by mfs-mindsize** [comment]
@pinksharpii - I took a swipe at this. A PR will be submitted soon, after an internal (team) review on my end. If you look in the method: remove_order_item(), the solution looks something like this:

```
$order = wc_get_order( $order_id );
$order->calculate_taxes( $calculate_tax_args );
$order->calculate_totals( false );

do_action( 'woocommerce_ajax_order_items_removed', $item_id, $item, $changed_stock, $order );

// Get HTML to return.
ob_start();
include __DIR__ . '/admin/meta-boxes/views/html-order-items.php';
$items_html = ob_get_clean();
```
** Extra lines pasted here for context. Only the do_action is new.

**Comment by mfs-mindsize** [comment]
FYI - PR submitted https://github.com/woocommerce/woocommerce/pull/28936

**Comment by jorgeatorres** [comment]
Fixed in #28936 & related #29241.
